adapters/data_parser.py

```diff
@@ -45,8 +46,45 @@ def read_scores(file_path: str) -> list[Student]:
     return results
 
 
-def display_scores(results: list[Student]) -> None:
-    print("*** Created at :", datetime.datetime.now().strftime('%Y-%m-%d %H:%M'))
+def general_summary(results: list[Student], output_path: str, save_as_txt: bool, save_as_pdf: bool,
```

Comment:
Single responsibility principle:

This function (with fixed name) should generate text data for all students only (and let's move it to `main.py` as it is definitely not part of an adapter). Then, outside of this function, should be the remaining logic (checking boolean flags and calling output generating functions from this module).


So, within business_logic module there should be a function `for example process`